implementation/src/main/java/io/smallrye/mutiny/groups/UniOnNull.java

```diff
@@ -1,20 +1,20 @@
 package io.smallrye.mutiny.groups;
 
-import static io.smallrye.mutiny.helpers.ParameterValidation.SUPPLIER_PRODUCED_NULL;
 import static io.smallrye.mutiny.helpers.ParameterValidation.nonNull;
 
 import java.util.NoSuchElementException;
+import java.util.Objects;
 import java.util.function.Supplier;
 
 import io.smallrye.mutiny.Uni;
 import io.smallrye.mutiny.infrastructure.Infrastructure;
 
 public class UniOnNull<T> {
 
-    private final Uni<T> upstream;
+    private final UniOnPredicate<T> uniOnPredicate;
 
     public UniOnNull(Uni<T> upstream) {
-        this.upstream = nonNull(upstream, "upstream");
+        this.uniOnPredicate = new UniOnPredicate<>(nonNull(upstream, "upstream"), Objects::isNull);
```

Comment:
I opted to keep `UniOnNull` and `UniOnNotNull` instead of replacing it entirely with `UniOnPredicate`, but tried to refactor it to use the new `UniOnPredicate` everywhere. I think it's better to keep the classes because we then keep the API of `ifNull()` and `ifNotNull()` the same. If we removed those classes entirely, we would introduce some new methods (for example all methods that are part of `ifNotNull()` would also be included in `ifNull()` and vice-versa), which would bloat up the existing API and introduce methods especially in the null case which don't really make sense. Because you don't really need a `transform` method if you know already you have a null object, you're not gonna use that as input for your mapping function anyway.